manager/integration/tests/test_snapshot_check.py

```diff
@@ -0,0 +1,409 @@
+    assert snapMap[snap2.name].name == snap2.name
+    assert snapMap[snap2.name].parent == snap1.name
+    assert snapMap[snap2.name].removed is False
+    assert snapMap[snap3.name].name == snap3.name
+    assert snapMap[snap3.name].parent == snap2.name
+    assert snapMap[snap3.name].removed is False
+
+
+def write_device_random_data(dev, size):
+    exec_cmd = ["dd", "if=/dev/urandom", "of=" + dev,
+                "bs=1M", "count=" + str(size)]
+    try:
+        subprocess.check_output(exec_cmd)
+    except subprocess.CalledProcessError as e:
+        print(e.output)
+        return False
+    return True
+
+
+def check_hash_calculated_and_with_immediate_hash(client, volume_name, snapshot_data_integrity):  # NOQA
+    """
+    1. Create and attach a volume
+    2. Create snapshots
+    3. Check snapshots' checksums are calculated and set and also
+       check the checksums of the snapshot disk files are matched
+       with the values from the api
+       Check snapshots' checksums are calculated by periodic snapshot
+       verification mechanism and set and also check the checksums of
+       the snapshot disk files are matched with the values from the api
+    """
+
+    # Step 1
+    volume = create_and_check_volume(client, volume_name, 3,
+                                     size=str(16 * Mi),
+                                     snapshot_data_integrity=snapshot_data_integrity) # NOQA
+
+    lht_hostId = get_self_host_id()
+    volume = volume.attach(hostId=lht_hostId)
+    volume = wait_for_volume_healthy(client, volume_name)
+
+    volume = client.by_id_volume(volume_name)
+
+    # Step 2
+    create_snapshots(client, volume, 1)
+
+    # Step 3
+    assert check_snapshot_checksums(volume)
+
+
+def check_hash_calculated_and_without_immediate_hash(client, volume_name, snapshot_data_integrity): # NOQA
+    """
+    1. Create and attach a volume
+    2. Create snapshots
+    3. Sleep for 120 seconds. Check snapshots' checksums are not calculated
+       and not set, because the immediate hash after snapshot creation feature
+       is disabled
+    4. Sleep for 180+60 seconds, check snapshots' checksums are calculated and
+       set by the periodic check mechanism
+    """
+
+    # Step 1
+    volume = create_and_check_volume(client, volume_name, 3,
+                                     size=str(16 * Mi),
+                                     snapshot_data_integrity=snapshot_data_integrity) # NOQA
+
+    lht_hostId = get_self_host_id()
+    volume = volume.attach(hostId=lht_hostId)
+    volume = wait_for_volume_healthy(client, volume_name)
+
+    volume = client.by_id_volume(volume_name)
+
+    # Step 2
+    create_snapshots(client, volume, 1)
+
+    # Step 3
+    time.sleep(120)
+    assert check_snapshot_checksums_not_set(volume)
+
+    # Step 4
+    # After 5 minutes, the periodic snapshot check should
+    # calculate the checksum and set it in the xattr
+    time.sleep(240)
+    assert check_snapshot_checksums(volume)
+
+
```

Comment:
Hi @derekbit :
     According to https://github.com/longhorn/longhorn/issues/4210#issuecomment-1286534849, I think we could add the function `wait_for_snapshot_checksums_gernerate`.
     This function that will print snapshot checksum generates time on the console. We also could restrict the range of periods instead of `time.sleep`. What do you think?
![Screenshot_20221126_002523](https://user-images.githubusercontent.com/104337648/204025200-c0c8ebd3-cd98-422b-a4e5-40f0674b6830.png)
![Screenshot_20221126_001659](https://user-images.githubusercontent.com/104337648/204025225-088eeb7b-bbd5-482b-8443-95433b3acb2d.png)

```suggestion
    # Step 3
    # Step 4
    # After 5 minutes, the periodic snapshot check should
    # calculate the checksum and set it in the xattr
    checksums_set, generate_checksums_time = wait_for_snapshot_checksums_gernerate(volume)
    time_range = range(120,360)
    assert checksums_set
    assert generate_checksums_time in time_range

def wait_for_snapshot_checksums_gernerate(volume_name):
    for i in range(RETRY_COUNTS*4):
        snapshot_checksums_not_set = check_snapshot_checksums_not_set(volume_name)
        if snapshot_checksums_not_set:
            time.sleep(RETRY_INTERVAL)
            continue
        else:
            print(f'checksums set in {i} sec')
            generate_checksums_time = i
            snapshot_checksums_generate = True
            break
    return snapshot_checksums_generate, generate_checksums_time
```

Reply:
LGTM. Thank you.

Reply:
Addressed. Thank you.